### PR TITLE
only start in fullscreen if enabled in Preferences or via command line arg

### PR DESCRIPTION
**Pull request by ronso0**
... by simply exiting fullscreen and maximizing the window if fullscreen is not enabled in Preferences.

The issue is that saveGeometry() also saves the fullscreen state in `[MainWindow] geometry` which is restored on next start regardless of the explicit configuration. Then Mixxx doesn't see a fullscreen flag thus the Fullscreen checkbox is not updated programmatically.
I didn't investigate how that would affect other actions in the View menu and can render those hotkeys unfunctional when restoring unrequested fullscreen mode, but but hopefully that's fixed, too.

https://bugs.launchpad.net/mixxx/+bug/1882474
https://bugs.launchpad.net/mixxx/+bug/1904135
https://bugs.launchpad.net/mixxx/+bug/1909485

one small issue remains:
when Mixxx is started in fullscreen after being shutdown in windowed mode the launch progress is in windowed mode.
when starting in fullscreen mode repeatedly the launch progress is also fullscreen.

### CODE REVIEW

**Comment by ronso0** [comment]
labeled 'Major bug' because in https://bugs.launchpad.net/mixxx/+bug/1882474 it was reported that previously View menu hotkeys would no work

**Comment by Holzhaus** [comment]
I didn't even notice this because I always fullscreen Mixxx using <Super+f> on Pop Shell (and i3 before that). In that case the fullscreen checkmark in the view menu is *never* set :smile: 

I just tested and noticed the following:
1. Start mixxx in windowed mode
2. Enable fulllscren via the View menu -> Checkmark is set and mixxx is in full screen mode
3. Close Mixxx via File -> Exit
4. Start Mixxx again -> Mixxx starts in full screen, but the checkmark in the view menu is *not* set

**Comment by ronso0** [comment]
Oh no, not that again. I remember there were issues with File > Exit vs. system command 'Close window'/window button X.
IIRC Mixxx crashed when using File > Exit. Are you sure Mixxx shuts down cleanly?

Can not confirm the fullscreen issue, though, neither with x/'close window' nor File > Exit. That's exactly was this is trying to fix ;)
Are you sure Mixxx doesnt

**Comment by uklotzde** [comment]
> I didn't even notice this because I always fullscreen Mixxx using <Super+f> on Pop Shell (and i3 before that). In that case the fullscreen checkmark in the view menu is _never_ set smile
> 
> I just tested and noticed the following:
> 
>     1. Start mixxx in windowed mode
> 
>     2. Enable fulllscren via the View menu -> Checkmark is set and mixxx is in full screen mode
> 
>     3. Close Mixxx via File -> Exit
> 
>     4. Start Mixxx again -> Mixxx starts in full screen, but the checkmark in the view menu is _not_ set

The check mark never worked for me after restarting Mixxx.

**Comment by ronso0** [comment]
Well, does it now?
which window manager do yo use?
for window managers that don't display title bars there is no difference between maximized and fullscreen, right?

The mentioned bugs only affect OS with 'regular' window decoration I guess.

**Comment by uklotzde** [comment]
Now the full-screen state is not remembered between restarts. The window is only maximized after restarting. The check mark in the menu is not set which is at least consistent.

But I tested after merging into main and resolving some conflicts with the CoreServices extraction, not 2.3 yet.

**Comment by ronso0** [comment]
> Now the full-screen state is not remembered between restarts. The window is only maximized after restarting.

That's correct. No fullscreen mode unless you check it in Preferences, or toggle it per session via command line.

**Review by uklotzde — APPROVED** [review]
Works for me on GNOME Shell.

**Comment by ronso0** [comment]
@daschuer Could yo try this with Ubuntu, please?

**Comment by Holzhaus** [comment]
> > Now the full-screen state is not remembered between restarts. The window is only maximized after restarting.
> 
> That's correct. No fullscreen mode unless you check it in Preferences, or toggle it per session via command line.

Ah okay. For me it was confusing because I use a tiling window manager and maximized is the same as full screen. 

What's the rationale for maximizing? Why not just open the window normally?

**Comment by ronso0** [comment]
I decided to 'maximize' in order to have a compromise between unrequested fullscreen and previously fullscreen, then starting in 'normal' window.
I guess it doesn't make much of a difference: either users know about the 'always fullscreen' option or they don't.

we could mention that option with a popup when users toggle fullscreen manually, incl. a checkbox "[  ] don't show that hint again"

**Comment by uklotzde** [comment]
Merging this now. If it does not work for someone we will get feedback soon, hopefully ;) Thank you! LGTM

**Pull request merged**
